docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+| thanos_delete_delay_seconds               | gauge   | Configured delete delay in seconds.                                                                                                                        |
+
+### List of Metrics Exported By Store API
+- The Thanos Sidecar component implements and exposes a gRPC [Store API](https://github.com/thanos-io/thanos/blob/main/docs/quick-tutorial.md#store-api); this implementation allows one to query the metric data stored in Prometheus.
+
+  | Metric Name                             | Type      | Description                                                |
+  |-----------------------------------------|-----------|------------------------------------------------------------|
+  | thanos_store_api_query_duration_seconds | histogram | Duration of the Thanos Store API select phase for a query. |
+
+### List of Metrics Exported By Alert
+- Alert sends alert notifications to Alertmanager clusters.
+- Queue is a queue of alert notifications waiting to be sent; the queue is consumed in batches and entries are dropped at the front if it runs full.
+
+  | Metric Name                              | Type      | Description                                                                    |
+  |------------------------------------------|-----------|--------------------------------------------------------------------------------|
+  | thanos_alert_queue_alerts_dropped_total  | counter   | Total number of alerts that were dropped from the queue.                       |
+  | thanos_alert_queue_alerts_popped_total   | counter   | Total number of alerts popped from the queue.                                  |
+  | thanos_alert_queue_alerts_pushed_total   | counter   | Total number of alerts pushed to the queue.                                    |
+  | thanos_alert_queue_capacity              | gauge     | Capacity of the alert queue.                                                   |
+  | thanos_alert_queue_length                | gauge     | Length of the alert queue.                                                     |
+  | thanos_alert_sender_alerts_dropped_total | counter   | Total number of alerts dropped in case of all sends to alertmanagers failed.   |
+  | thanos_alert_sender_errors               | counter   | Total number of errors while sending alerts to alertmanager.                   |
+  | thanos_alert_sender_alerts_sent_total    | counter   | Total number of alerts sent by alertmanager.                                   |
+  | thanos_alert_sender_errors_total         | counter   | Total number of errors while sending alerts to alertmanager.                   |
+  | thanos_alert_sender_latency_seconds      | histogram | Latency for sending alert notifications (not including dropped notifications). |
```

Comment:
These here seem to be metrics exported by the Thanos Ruler, no? I don't think there are other components of Thanos exposing these metrics because only the ruler talks to alertmanager.

Reply:
I will move the thanos_alert_* metrics to the Thanos Ruler PR accordingly.